.github/workflows/ci.yml

```diff
@@ -29,6 +29,7 @@ jobs:
         if: startsWith(matrix.rust, 'nightly')
         run: cargo check -Z features=dev_dep
       - run: cargo test
+      - run: cargo build --no-default-features
```

Comment:
x86_64-unknown-linux-gnu can link to std, so this does not ensure that no-std is supported.
Also, note that [cargo v1 resolver merges features of dev-dependencies and normal dependency](https://doc.rust-lang.org/cargo/reference/features.html#feature-unification).


To work around these problems, you need a tool like cargo-hack that can work around the v1 resolver problems, and a real no-std target.

```yaml
      - name: Install cargo-hack
        uses: taiki-e/install-action@cargo-hack
      - run: rustup target add thumbv7m-none-eabi
      - run: cargo hack build --target thumbv7m-none-eabi --no-default-features --no-dev-deps


```

Reply:
(Well, we should also fix other no-std crates in smol-rs)